test/util/util.js

```diff
@@ -31,4 +31,46 @@ describe('util', function() {
       expect(b).to.eql({x: 7, z: {}});
     });
   });
+
+  describe('promisify', function() {
+    it('wrapped functions return promise', async function() {
+      var targetFn = function(num, cb) {
+        setImmediate(function() {
+          cb(undefined, num);
+        });
+      };
+      var promisedFn = util.promisify(targetFn);
+      var promise = promisedFn(3);
+      expect(promise).to.be.instanceOf(Promise);
+      var result = await promise;
+      expect(result).to.equal(3);
+    });
+
+    it('wrapped functions throw errors passed to callback', async function() {
+      var targetFn = function(num, cb) {
+        setImmediate(function() {
+          cb(new Error(`Error ${num}`));
+        });
+      };
+      var promisedFn = util.promisify(targetFn);
+      try {
+        await promisedFn(3);
+      } catch (error) {
+        expect(error).to.have.property('message', 'Error 3');
+      }
```

Comment:
We should also ensure the test case will fail if the function resolves. One way is to add this after the `await`:
```
fail('Expected promisedFn to reject, but it successfully resolved');
```

Same comment on the next test case, too.

Reply:
oh, good call 👍